scripts/datasets/update_download_stats.py

```diff
@@ -0,0 +1,107 @@
+                            if chunk:  # filter out keep-alive new chunks
+                                m.update(chunk)
+                                f_size += len(chunk)
+                        hex_hash = m.hexdigest()
+                        file_size = f_size
+                        end = time.time()
+                        print('{}={}, size={}, Time spent={}'.format(algorithm, hex_hash, file_size,
+                                                                     end - start))
+                        if cache is None:
+                            cache = OrderedDict()
+                        cache[url] = (hex_hash, file_size)
+                        if save_path is not None:
+                            with open(save_path, 'a', encoding='utf-8') as of:
+                                of.write('{} {} {}\n'.format(url, hex_hash, file_size))
+                        break
+                    except Exception as e:
+                        retries -= 1
+                        if retries <= 0:
+                            raise e
+                        print('download failed due to {}, retrying, {} attempt{} left'
+                              .format(repr(e), retries, 's' if retries > 1 else ''))
+                return obj, (hex_hash, file_size)
+            else:
+                return obj
+        elif isinstance(obj, tuple):
+            return tuple((_get_hash_and_size(ele, retries, algorithm, cache, save_path)
+                          for ele in obj))
+        elif isinstance(obj, list):
+            return [_get_hash_and_size(ele, retries, algorithm, cache, save_path) for ele in obj]
+        elif isinstance(obj, dict):
+            return {k: _get_hash_and_size(v, retries, algorithm, cache, save_path)
+                    for k, v in obj.items()}
+        else:
+            return obj
+    if cache is None:
+        cache = OrderedDict()
+    else:
+        cache = copy.deepcopy(cache)
+    if save_path is not None and os.path.exists(save_path):
+        with open(save_path, 'r', encoding='utf-8') as f:
+            for line in f:
+                line = line.strip()
+                url, hex_hash, file_size = line.split()
+                cache[url] = (hex_hash, file_size)
+    _get_hash_and_size(obj, retries, algorithm, cache, save_path)
+    return obj, cache
+
+
+if __name__ == '__main__':
+    get_hash_and_size([wmt_para_urls, wmt_mono_urls],
+                      save_path=os.path.join(_CHECK_SUM_BASE, 'wmt.txt'))
+    get_hash_and_size(squad_urls,
+                      save_path=os.path.join(_CHECK_SUM_BASE, 'squad.txt'))
+    get_hash_and_size(lm_urls,
+                      save_path=os.path.join(_CHECK_SUM_BASE, 'language_model.txt'))
+    get_hash_and_size(midi_urls,
+                      save_path=os.path.join(_CHECK_SUM_BASE, 'music_midi.txt'))
+    get_hash_and_size(book_urls,
+                      save_path=os.path.join(_CHECK_SUM_BASE, 'book_corpus.txt'))
```

Comment:
What's the point of this file?

Reply:
For easy update of the download statistics. I was thinking to also include it as a test.